src/pages/index.js

```diff
@@ -1,68 +1,204 @@
+      return { ...data, isLiked: false };
+    });
+  }
+};
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__description",
+  avatarSelector: "#profile-image",
 });
 
 const renderCard = (item) => {
-  const card = new Card(item, "#card-template", handleImageClick);
+  const card = new Card(
+    {
+      name: item.name,
+      link: item.link,
+      alt: item.alt,
+      _id: item._id,
+      likes: item.likes,
+      isLiked: item.isLiked,
+    },
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    userInfo
+  );
   const cardElement = card.getView();
   cardList.addItem(cardElement);
 };
 
 const cardList = new Section(
   {
-    items: initialCards.reverse(),
+    items: [],
     renderer: renderCard,
   },
   ".cards__list"
 );
 
-cardList.renderItems();
+// Extend PopupWithForm for loading state handling
+class CustomPopupWithForm extends PopupWithForm {
+  setLoadingState(isLoading) {
+    const saveButton = this._popup.querySelector(".modal__button");
+
+    if (isLoading) {
+      saveButton.textContent = "Saving...";
+      saveButton.disabled = true;
+    } else {
+      saveButton.textContent = "Save";
+      saveButton.disabled = false;
+    }
+  }
+}
+
+const addCardPopup = new CustomPopupWithForm("#modal-add-card", (formData) => {
+  addCardPopup.setLoadingState(true);
 
-const addCardPopup = new PopupWithForm("#modal-add-card", (formData) => {
-  renderCard({ name: formData.title, link: formData.url, alt: formData.title });
-  addCardPopup.close();
+  api
+    .addCard({ name: formData.title, link: formData.url })
+    .then((data) => {
+      renderCard(data);
+      addCardPopup.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      addCardPopup.setLoadingState(false);
+    });
+});
+
+const editPfpPopup = new CustomPopupWithForm("#modal-edit-pfp", (formData) => {
+  editPfpPopup.setLoadingState(true);
+
+  api
+    .updateAvatar({ avatar: formData.url })
+    .then((data) => {
+      userInfo.setUserAvatar(data.avatar);
+      editPfpPopup.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      editPfpPopup.setLoadingState(false);
+    });
+});
+
+editPfpPopup.setEventListeners();
+
+profileEclipse.addEventListener("click", () => {
+  editPfpPopup.open();
 });
 
 addCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#modal-edit-profile",
-  (formData) => {
-    userInfo.setUserInfo(formData.name, formData.description);
-    editProfilePopup.close();
-  }
-);
+const editProfilePopup = new CustomPopupWithForm("#modal-edit-profile", (formData) => {
+  editProfilePopup.setLoadingState(true);
+
+  api
+    .updateUserInfo({ name: formData.name, about: formData.description })
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      editProfilePopup.close();
+    })
+    .catch((err) => console.error(err))
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors